### DISCUSSION TITLE
Using Gatsby Cloud CMS previews and Gatsby source shopify

### DISCUSSION THREAD

**Discussion by moreguppy** [gatsbyjs/gatsby#32416] [Help]
Is there a best practice for using Gatsby Cloud CMS Preview with gatsby-source-shopify?

I'm aware of the limitations for the source plugin which is using the "bulk requests" API — and only one of these operations can run at a given time.

Does this mean we won't be able to use CMS preview for a gatsby shopify site, since the preview update and published update will always run simultaneously and cancel each other out?

**Comment by DanielSLew** [comment]
Hi @moreguppy,

Currently we use a work-around for the limitations of the bulk requests API. We consider production builds "priority builds", so a production build will cancel a preview using the bulk API request, but a preview won't cancel a production build using the bulk api, it will just wait until the production build finishes its bulk API request.

Because of this work-around, you will be able to use CMS preview, but the builds will have to wait for one another, however they should be fairly quick on warm builds because of incremental updates.

Another thing to note is that at the moment we don't fully support Preview for Shopify, the production build and the preview build are the same builds. They both pull published content. So it's not a "true" preview (previewing draft content rather than published content). We will be adding full preview capability later this year.

Thanks for your interest in our Shopify plugin, let me know if this answers all your questions.

Have a great day!

**Reply by moreguppy** [reply]
Hi @DanielSLew, thank you for the detailed reply, super helpful!
